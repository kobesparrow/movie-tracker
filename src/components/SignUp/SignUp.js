import React from 'react';

const SignUp = ({name, password, email, handleChange, handleSubmit, userState}) => {
  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label>
          Name
          <input
            type='text'
            name='name'
            placeholder='Name'
            value={name}
            onChange={handleChange}
          />
        </label>
        <label>
          E-mail
          <input
            type='email'
            name='email'
            placeholder='example@example.com...'
            value={email}
            onChange={handleChange}
          />
        </label>
        <label>
          Password
          <input
            type='password'
            name='password'
            placeholder='Password'
            value={password}
            onChange={handleChange}
          />
        </label>
        <input
          type='submit'
          value='Sign Up'
        />
        <button onClick={() => userState('initial')}>GO BACK</button>
      </form>
    </div>
  )
}

export default SignUp;
