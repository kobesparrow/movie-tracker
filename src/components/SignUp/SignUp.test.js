import React from 'react';
import SignUp from './SignUp';
import { shallow } from 'enzyme';

let mockState = {
    name: '',
    password: '',
    email: ''
}

let mockChange = jest.fn();
let mockSubmit = jest.fn();

describe('Login', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <SignUp
                {...mockState}
                handleSubmit={mockSubmit}
                handleChange={mockChange}
          />
        )
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
});