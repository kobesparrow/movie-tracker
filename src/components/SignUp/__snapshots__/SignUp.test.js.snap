// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<div>
  <form
    onSubmit={[MockFunction]}
  >
    <label>
      Name
      <input
        name="name"
        onChange={[MockFunction]}
        placeholder="Name"
        type="text"
        value=""
      />
    </label>
    <label>
      E-mail
      <input
        name="email"
        onChange={[MockFunction]}
        placeholder="example@example.com..."
        type="email"
        value=""
      />
    </label>
    <label>
      Password
      <input
        name="password"
        onChange={[MockFunction]}
        placeholder="Password"
        type="password"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
